Source/SeaInk.Core/Entities/Tables/GoogleTable.cs

```diff
@@ -0,0 +1,488 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Services;
+using Google.Apis.Sheets.v4;
+using Google.Apis.Sheets.v4.Data;
+using Google.Apis.Util.Store;
+using SeaInk.Core.Models.Tables;
+using SeaInk.Core.Models.Tables.Exceptions;
+using SeaInk.Core.Models.Tables.Enums;
+using LineStyle = SeaInk.Core.Models.Tables.Enums.LineStyle;
+
+namespace SeaInk.Core.Entities.Tables
+{
+    /// <inheritdoc cref="ITable"/>
+    public class GoogleTable : ITable
+    {
+        private static readonly string[] Scopes = {SheetsService.Scope.Spreadsheets};
+        private const string ApplicationName = "Sea Ink";
+
+        private string SpreadsheetId { get; set; } = "";
+        private SheetsService Service { get; set; }
+
+        public int SheetCount => GetSpreadsheet().Sheets.Count;
+
+
+        public int ColumnCount(TableIndex index)
+        {
+            int? count = GetSpreadsheet().Sheets[index.SheetId].Properties.GridProperties.ColumnCount;
+
+            if (!count.HasValue)
+                throw new NonExistingIndexException();
+
+            return count.Value;
```

Comment:
Кажется, можно ограничиться
```suggestion
            return count ?? throw new NonExistingIndexException();
```

Reply:
Не знал что так можно, выглядит красиво